src/containers/Plot/index.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import PlotlyPlot from "react-plotly.js";
+// import { ActionCreator } from "redux";
+
+import { State } from "../../state/types";
+import { PlotConfig } from "../ResultsPanel/types";
+import { getCurrentTime } from "../../state/selection/selectors";
+// import { changeTime } from "../../state/selection/actions";
+// import { ChangeTimeAction } from "../../state/selection/types";
+import { PLOT_STYLE } from "../ResultsPanel/constants";
+
+interface PlotProps {
+    plotConfig: PlotConfig;
+    time: number;
+    // changeTime: ActionCreator<ChangeTimeAction>;
+}
+
+const styles = require("./style.css");
+
+class Plot extends React.Component<PlotProps, {}> {
+    shouldComponentUpdate() {
+        return this.props.plotConfig.hasTimeIndicator;
```

Comment:
I'm curious if this acts the same as `return "prevTime !== currentTime" `. Did you try that? I dont think plot config is ever changing otherwise, and it's the only other prop, so it I think it would act the same

Reply:
We need to factor `hasTimeIndicator` into this method, because otherwise this plot component would try to update at every frame even if it doesn't have a time indicator (due to it receiving a dynamic `time` prop from redux). To double check, I just tried out `return this.props.time !== nextProps.time;` (equivalent to your suggestion I think) and we went back to 3 fps with the Actin Bending Angles scatter plot taking a ton of time to update at every frame even though it doesn't have a time indicator. Let me know if I totally missed what you're saying 😆

Reply:
sorry, i meant `return this.props.plotConfig.hasTimeIndicator && this.props.time !== nextProps.time`; that it should explicitly check for the prop that matters updating. Also, I'm curious how this works when you change between trajectories. Does this component end up remounting, or rerendering? If it remounts with new trajectory, you could just add the time plot on component mount, instead of having to check for it every update.

Reply:
Ah, ok! Yeah they should be equivalent because, like you said, `time` is the only thing that changes after initial mount. I just tried changing trajectories under both scenarios and the entire ResultsPanel and its children all remount. I like the suggestion in your last sentence, will implement

Reply:
Uh, not actually sure if ResultsPanel remounts or re-renders, but will check

Reply:
I verified that, when a new trajectory is loaded, ResultsPanel re-renders and the Plot components re-mount. So I moved the initial creation of the time indicator to `componentDidMount` :)